pkg/provider/aws/resources/elasticache.go

```diff
@@ -64,35 +59,95 @@ func (ecsn *ElasticacheSubnetgroup) Id() core.ResourceId {
 	}
 }
 
-func CreateElasticache(cfg *config.Application, dag *core.ResourceGraph, source core.Construct) *ElasticacheCluster {
-	ec := &ElasticacheCluster{
-		Name:            elasticacheClusterSanitizer.Apply(fmt.Sprintf("%s-%s", cfg.AppName, source.Provenance().ID)),
-		Engine:          "redis", // TODO determine this from the type of `source`
-		CloudwatchGroup: NewLogGroup(cfg.AppName, fmt.Sprintf("/aws/elasticache/%s-%s-persist-redis", cfg.AppName, source.Id()), source.Provenance(), 0),
-		SubnetGroup: &ElasticacheSubnetgroup{
-			Name:          elasticacheClusterSanitizer.Apply(fmt.Sprintf("%s-%s", cfg.AppName, source.Provenance().ID)),
-			Subnets:       GetSubnets(cfg, dag), // TODO when we allow for segmented networks, need to determine which network (subnets) this lives in
-			ConstructsRef: []core.AnnotationKey{source.Provenance()},
+type ElasticacheClusterCreateParams struct {
+	AppName string
+	Refs    []core.AnnotationKey
+	Name    string
+}
+
+func (ec *ElasticacheCluster) Create(dag *core.ResourceGraph, params ElasticacheClusterCreateParams) error {
+	ec.Name = aws.ElasticacheClusterSanitizer.Apply(fmt.Sprintf("%s-%s", params.AppName, params.Name))
+	ec.ConstructsRef = params.Refs
+	ec.NumCacheNodes = 1
+	ec.NodeType = "cache.t3.micro"
+	ec.Engine = "redis"
```

Comment:
these should be set in the configure if they are allowed to be configured, otherwise via an edge. The reason why we dont want to do them in a create is because multiple things may call create and all have a different context as to what this is going to be used for and we wouldnt catch conflicts

Reply:
I've updated this to follow the basic pattern used by the `ORM` and `KV` constructs.